[RFR] Add invite to create the first record when the list is empty (#4065)
marmelab/react-admin: docs/List.md

* [`filterDefaultValues`](#filter-default-values) (the default values for `alwaysOn` filters)
* [`pagination`](#pagination)
* [`aside`](#aside-component)

> * [`emptyState`](#empty-state)

Reviewer: I'd call it `empty` instead

```
{% endraw %}

### Empty state

> When there is no result, and there is no active filter, and the resource has a create page then a special page inviting the user to create the first record is displayed. Use the `emptyState` prop to modify that page, passing your custom component:

Reviewer: ```suggestion
When there is no result, and there is no active filter, and the resource has a create page, react-admin displays a special page inviting the user to create the first record.

You can use the `emptyState` prop to replace that page by a custom component:
```

{% raw %}
```jsx
import Button from '@material-ui/core/Button';
import { CreateButton, List } from 'react-admin';

const EmptyState = ({ basePath }) => (
    <div style={{ textAlign: 'center', margin: '1em' }}>

>         <h1>No products available</h1>

Reviewer: I'd use `resource` in the example

        <div>Create one or import from a file</div>
        <CreateButton basePath={basePath} />
        <Button onClick={...}>Import</Button>
    </div>
);

const ProductList = props => (
    <List emptyState={<EmptyState />} {...props}>
        ...
    </List>
);
```
{% endraw %}

> The `emptyState` component receives the same props as the `aside` prop. Read the [section above](#aside-component) to check them. 

Reviewer: no, please repeat them - developers don't read the docs sequentially.
